library/hawkular_alerts_group_trigger.py

```diff
@@ -155,49 +192,70 @@ def required_updates(self, trigger, name, severity, enabled):
+            condition.context = {'name': name}
+            gc.addCondition(condition)
+        try:
+            self.client.create_group_conditions(group_id, conditions["trigger_mode"], gc)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to set group trigger conditions. Error: {error}".format(error=e))
+        self.changed = True
+
+    def update_group_trigger(self, trigger, updates, conditions):
         """ Updates a group Trigger in Hawkular Alerts
 
             Returns:
                 whether or not a change took place and a short message
                 describing the operation executed
         """
+        for attr in updates:
+            setattr(trigger, attr, updates[attr])
         try:
-            for attr in updates:
-                setattr(trigger, attr, updates[attr])
             self.client.update_group_trigger(trigger.id, trigger)
-            self.changed = True
-            return dict(
-                msg="Successfully updated group trigger {group_id}".format(group_id=trigger.id),
-                changed=self.changed)
         except Exception as e:
             self.module.fail_json(msg="Failed to update group trigger. Error: {error}".format(error=e))
+        if conditions:
+            if self.conditions_update_required(trigger.id, conditions['conditions']):
+                self.set_group_trigger_conditions(trigger.id, conditions)
+        self.changed = True
+        return dict(
+            msg="Successfully updated group trigger {group_id}".format(group_id=trigger.id),
+            changed=self.changed)
 
-    def create_group_trigger(self, group_id, name, severity, enabled):
+    def create_group_trigger(self, group_id, name, severity, enabled, conditions):
         """ Creates a group Trigger in Hawkular Alerts
 
             Returns:
                 whether or not a change took place and a short message
                 describing the operation executed
         """
-        try:
-            #  create trigger object
-            trigger = hawkular.alerts.Trigger()
-            trigger.id = group_id
-            trigger.name = name
-            trigger.severity = severity
-            trigger.enabled = enabled
+        #  create trigger object
+        trigger = hawkular.alerts.Trigger()
+        trigger.id = group_id
+        trigger.name = name
+        trigger.severity = severity
+        trigger.enabled = enabled
 
+        try:
             self.client.create_group_trigger(trigger)
-            self.changed = True
-            return dict(
-                msg="Successfully created group trigger {group_id}".format(group_id=group_id),
-                changed=self.changed)
         except Exception as e:
             self.module.fail_json(msg="Failed to create group trigger. Error: {error}".format(error=e))
 
+        if conditions:
+            self.set_group_trigger_conditions(group_id, conditions)
+        self.changed = True
```

Comment:
Probably doens't matter, but it feels like `changed = True` belongs after `self.client.create_group_trigger()`.
`set_group_trigger_conditions()` does is own `changed = True`.